feat: create payments for all companies from portal

HKuz opened the pull request:
Closes #56 

When multiple companies exist in ERPNext, this changes the behavior between the portal-side adding a new payment method vs from the desk side in the Electronic Payments dialog:

- [x] New payment method from Portal: cycles over all Electronic Payment Settings docs and adds the method for each. The display table of Payment Methods adds a Company field to clarify why the method may show up multiple times
- [x] Desk side: only adds the method for the Company the PI/PO/SI/SO is linked to. When displaying Saved payment methods in the dialog, it only shows methods available for that Company
- [x] update documentation

Other clean-up:
- [x] adds the `sending_use_clearing_account` selection for Sending Payments tab in EP Settings (previously, we had it so accepting payments were required and it just used that field)
- [x] in event that EP Settings has the "automatically create portal payment methods" box UN-checked, use left join to collect payment methods for the dialog box
- [x] re-work how to retrieve the Customer ID (provider's profile ID for a party) - currently it's saved in a field on the party doc (Customer / Supplier), but if different companies in ERPNext use different API keys or different providers completely, this value should be taken from the Electronic Payment Profile

HKuz committed:
feat: create payments for all companies from portal

HKuz committed:
feat: update customer profile to retrieve from EPP

HKuz commented:
How I tested:

- created a new company (Ambrosia Pie Company)
- created Electronic Payment Settings for new company with different providers - Chelsea Fruit Co uses Authorize/Mercury and Ambrosia Pie Co uses Stripe/Wise
- Log into portal as a Supplier - add a new ACH (use non wire routing number like 226077642 - if don't, it causes error in Wise because the API request has to be different for a Wire-friendly routing no). See that the new payment method added for both companies. Delete one.
- Log into desk as Admin, go to PI for company with payment method - should see as option. Go to PI for other company, should not see any saved payment methods
- Repeat for Customer - if creating new ACH, should NOT see it create a payment method for the company using Stripe as provider. If using Card, the test card number 4111111111111111 works for both providers. Make a payment on an invoice from both Portal side as customer, or desk side as admin

HKuz committed:
feat: add company filter to account fields in EP Settings

HKuz committed:
docs: update screen shots, remove company field, explain portal behavior

fproldan commented:
@HKuz I'm having an error [here](https://github.com/agritheory/electronic_payments/pull/60/files#diff-3c8b8c4a1c4c7af5a4ee29e74db8a5cb0700902e17bada64e40378688edf44dcR176) since `electronic_payment_profile` field in `Customer` and `Supplier` was deleted.

Do you know what would be the equivalent now?

HKuz commented:
@fproldan - sure thing - the short answer is that code should be grabbing the provider's party profile ID (`party_profile` field) from the Electronic Payments Profile doc (it was already a field there) for a given payment method vs the (now deleted) field on the party doc itself. If you don't have a specific payment method, you can check for that field in any EPP doc using the same provider gateway (caution - the gateway field uses "Authorize" not "Authorize.net"). Worst case, we can add an API call to the provider to check if there's already a profile ID in the event the party was in the system but deleted all their methods.

The background behind why we removed the field is because a client using the app has multiple companies in their ERPNext instance and each company has their own set of API keys for the provider (so one Party may have the exact same payment method saved with 2+ companies, so they'd have different profie IDs and payment profile tokens with each company's provider account). While Mercury and Wise don't use a party profile ID when setting up a new payment method for a Supplier, Authorize and Stripe do use that. This isn't an issue when profiles are added from an Order/Invoice, since the doc has a Company field, but it is when adding from the portal (as I'm sure you remember!). I initially tried solving the issue by putting a Company field on the Party doc to know which Company they did business with, but client feedback asked us to instead add the payment method for all companies with an Electronic Payments Settings doc configured.
